storm-client/src/jvm/org/apache/storm/executor/Executor.java

```diff
@@ -462,6 +473,30 @@ protected void setupMetrics() {
                     }
                 }
             );
+            if (interval == v2MetricsTickInterval) {
+                v2TickScheduled = true;
+            }
+        }
+
+        if (!v2TickScheduled) {
+            LOG.info("Scheduling v2 metrics tick for interval {}", v2MetricsTickInterval);
+            StormTimer timerTask = workerData.getUserTimer();
```

Comment:
This code looks similar to above. Can we de-duplicate?

Reply:
sure, was debating same thing as I committed.

Reply:
updated